src/task/state.rs

```diff
@@ -165,14 +166,14 @@ impl Output {
     }
 
     /// Construct an [`Output`]` with an error message.
-    pub fn error(msg: String) -> Self {
-        Self::Err(msg)
+    pub fn error(code: Option<i32>, msg: String) -> Self {
```

Comment:
@kdesjard This PR does not make sense for normal outputs when someone is not running Shell commands. Rust has rich error-handling features, which are much better than exit codes.

Also, this method is changed, which is a breaking change for all the users of Dagrs. We have to revert this change adding a new status field like `ErrWithExitCode`